### DISCUSSION TITLE
Wind/Rotor misalignment and estimated wind speed

### DISCUSSION THREAD

**Discussion by ghuwart** [NatLabRockies/ROSCO#317] [Q&A]
Dear all,

I am running some misaligned simulations with ROSCO and TSR tracking (`VS_ControlMode=2`).
I am using the filtered hub wind speed as the wind speed estimtor (`WE_Mode=0`) even though I've seen it is recommended to use at least the mode 1 as advised in #147 but my point is something else.

From my understanding, the estimated rotor speed should be the wind speed facing the rotor as it will be component that contributes the most to the motion of the turbine. Nevertheless, the wind speed read in `avrSWAP(27)` is the horizontal wind speed, _i.e._, the norm of the wind speed in the horizontal plane. That is what I saw in the OPENFAST code and this is what is read directly in mode 0 of the Wind speed estimator ([see here](https://github.com/NREL/ROSCO/blob/3ebe819411d5fadf630287d20bf44ff572149c63/rosco/controller/src/ControllerBlocks.f90#L353) in ControllerBlocks.f90).

When running with a misalignment, this can introduce undesired effects. In some of my cases, such as gust with extreme change of direction, the difference between the wind speed facing the rotor (around 6 m/s) and the norm (around 22 m/s) is hugely impactful since the control points are not the ones desired.

I reckon this is a fringe case but I think even standard production cases with misalignment could be affected with a 1 or 2% error on the "correct" wind speed for the TSR tracking.

Is my understanding of the situation correct?

Could we use `avrSWAP(24)`, _i.e._, the measured yaw error to get the component of `avrSWAP(27)` facing the rotor?

Thank you.

**Comment by dzalkind** [comment]
Your understanding is correct.  Ideally the projection of the wind in the direction facing the rotor would be used to determine the wind speed set points.  

If you're able to make the changes to the source code and recompile, I would encourage that for the quickest functionality you desire.  If you find something that works and would like to submit a PR of your changes, I can help with all the other details.

We are also planning developments to the controller to better support all DLCs in the next few months.  I opened #318 so we also keep track of this case.  Thanks.

Best, Dan

**Comment by ghuwart** [comment]
Great, thank you for the quick response as always. I will try to compile my own version to see if I can make it work for my case and will revert back to you.

I am happy to read that you are tackling the DLCs, good luck with that and I will gladly test the new functionalities.

Thanks once more.

**Comment by ghuwart** [comment]
Alright, a quick test with `LocalVar%HorWindV = cos(avrSWAP(24)) * avrSWAP(27)` instead of `LocalVar%HorWindV =  avrSWAP(27)` in [here](https://github.com/NREL/ROSCO/blob/3ebe819411d5fadf630287d20bf44ff572149c63/rosco/controller/src/ReadSetParameters.f90#L48) seems promising to me. It did give me the expected wind speed on my debugging model. I need to confirm my findings on full cases. I will not be able to continue investigating right away but I will come back to it later.
